packages/odata-common/src/payload-value-converter.ts

```diff
@@ -111,3 +111,35 @@ export const serializersCommon: EdmTypeMapping = {
   'Edm.String': identity,
   'Edm.Any': identity
 };
+
+/**
+ * @hidden
+ */
+export function createEdmToTs<V extends EdmTypeCommon>(deserializers: {
```

Comment:
For my understanding, shouldn't `V` also extend from `ExclusiveEdmTypeV2 | ExclusiveEdmTypeV4` or just from `EdmTypeShared<'any'>` for that matter ? . Same in `createUriConverter` below. All tests work fine though, so I might be mistaken here.

Reply:
EdmTypeShared contains EdmTypeCommon + ExclusiveEdmTypeV2 for V2. So I think to use `EdmTypeCommon` as a base type is fine. In principle you are right and the keys should be either the all V2 or all V4 ones. Not less like it would be possible now but also not more which would be the case if you use the EdmTypeShared<'any'>. Note EdmTypeShared is currently not select the version specif values.